pkg/api/rest/update.go

```diff
@@ -37,9 +37,16 @@ type RESTUpdateStrategy interface {
 	// the object.  For example: remove fields that are not to be persisted,
 	// sort order-insensitive list fields, etc.
 	PrepareForUpdate(obj, old runtime.Object)
-	// ValidateUpdate is invoked after default fields in the object have been filled in before
-	// the object is persisted.
+	// ValidateUpdate is invoked after default fields in the object have
+	// been filled in, but before the object is persisted.  This method is
+	// responsible for ensuring that any disallowed changes are flagged.
+	// For example: changing immutable fields.  It does not need to
+	// re-validate the whole object - that is what Validate() is for.
 	ValidateUpdate(ctx api.Context, obj, old runtime.Object) fielderrors.ValidationErrorList
+	// Validate is invoked after default fields in the object have been
+	// filled in and after the update itself has been validated, but before
+	// the object is persisted.
+	Validate(ctx api.Context, obj runtime.Object) fielderrors.ValidationErrorList
```

Comment:
I'm not really in favor of this.  If everyone has to implement ValidateUpdate, why force them to specify both?  Strategy#ValidateUpdate should invoke both methods - this is just an arbitrary distinction.